mkosi/distributions/gentoo.py

```diff
@@ -10,189 +8,97 @@
-        self.root = self.state.root
-        ret = self.try_import_portage()
-
-        from portage.package.ebuild.config import config as portage_cfg  # type: ignore
-
-        self.portage_cfg = portage_cfg(config_root=str(state.root),
-                                       target_root=str(state.root),
-                                       sysroot=str(state.root), eprefix=None)
-
-        PORTAGE_MISCONFIGURED_MSG = "Missing defaults for portage, bailing out"
-        # we check for PORTDIR, but we could check for any other one
-        if self.portage_cfg['PORTDIR'] is None:
-            die(PORTAGE_MISCONFIGURED_MSG)
-
-        self.profile_path = state.root / ret["profile_path"]
-        self.custom_profile_path = state.root / ret["custom_profile_path"]
-        self.ebuild_sh_env_dir = state.root / ret["ebuild_sh_env_dir"]
-        self.portage_cfg_dir = state.root / ret["portage_cfg_dir"]
-
-        self.portage_cfg_dir.mkdir(parents=True, exist_ok=True)
-
-        self.arch, _ = state.installer.architecture(state.config.architecture)
-        self.arch_profile = Path(f"default/linux/{self.arch}/{state.config.release}/no-multilib/systemd/merged-usr")
-        self.pkgs['sys'] = ["@world"]
-
-        self.pkgs['boot'] = [
-            "sys-kernel/installkernel-systemd-boot",
-            "sys-kernel/gentoo-kernel-bin",
-        ]
+    def kernel_image(name: str, architecture: Architecture) -> Path:
+        kimg_path = {
+            Architecture.x86_64: "arch/x86/boot/bzImage",
+            Architecture.arm64: "arch/arm64/boot/Image.gz",
+            Architecture.arm: "arch/arm/boot/zImage",
+        }[architecture]
 
-        self.emerge_vars = {
-            "FEATURES": " ".join(self.portage_features),
-            "USE": " ".join(self.portage_use_flags),
-        }
+        return Path(f"usr/src/linux-{name}") / kimg_path
 
-        self.fetch_fix_stage3()
-        self.set_useflags()
-        self.mkosi_conf()
-        self.get_snapshot_of_portage_tree()
-        self.update_stage3()
-        self.depclean()
+    @classmethod
+    def install(cls, state: MkosiState) -> None:
+        emerge_vars = {
+            "FEATURES": " ".join([
+                    # -user* are required for access to USER_CONFIG_PATH
+                    "-userfetch",
+                    "-userpriv",
+                    "-usersync",
+                    "-usersandbox",
+                    "-sandbox",
+                    "-pid-sandbox",  # for cross-compile scenarios
+                    "-network-sandbox",
+                    "parallel-install",
+                    "getbinpkg",
+                    "-candy",
+                    *(["noman", "nodoc", "noinfo"] if state.config.with_docs else []),
+            ]),
+            "USE": "initramfs minimal symlink"
+        }
+        arch = state.installer.architecture(state.config.architecture)
 
-    def fetch_fix_stage3(self) -> None:
         """usrmerge tracker bug: https://bugs.gentoo.org/690294"""
```

Comment:
This should be a comment now, since right now it's an unused string in the middle of a function.

Reply:
good catch, removed.